src/jump_input.jl

```diff
@@ -0,0 +1,67 @@
+using BilevelJuMP
+using Ipopt
+using JuMP
+using Test
+
+model = BilevelModel(Ipopt.Optimizer, mode = BilevelJuMP.ProductMode(1e-9))
+
+
+@variable(Upper(model), y, start = 8 / 15)
+@variable(Lower(model), x, start = 3.5 * 8 / 15)
+@objective(Upper(model), Min, 3x + y)
+@constraints(Upper(model), begin
+    u1, x <= 5
+    u2, y <= 8
+    u3, y >= 0
+end)
+
+@objective(Lower(model), Min, -x)
+@constraint(Lower(model), l1,  x +  y <= 8)
+@constraint(Lower(model), l2, 4x +  y >= 8)
+@constraint(Lower(model), l3, 2x +  y <= 13)
+@constraint(Lower(model), l4, 2x - 7y <= 0)
+
+
+function _build_single_model(
+    model::BilevelModel,
+)
+    upper = JuMP.backend(model.upper)
+    lower = JuMP.backend(model.lower)
+    link = JuMP.index(model.link)
+
+
+    return _build_single_model(upper, lower, link)
+end 
+
+function _build_single_model(
+    upper::MOI.ModelLike, 
+    lower::MOI.ModelLike, 
+    # A dictionary that maps variables in the upper to variables in the lower
+    link::Dict{MOI.VariableIndex,MOI.VariableIndex},
+)
+    # A new model to build
+    model = MOI.Utilities.Model{Float64}()
+    # Create a copy of the upper model
+    index_map = MOI.copy_to(model, upper)
+    # TODO ... add the lower problem to `model`
+    return model, upper_variables, lower_variables, lower_objective
+end
+
+
```

Comment:
Rather than testing the solution of the model, you should be testing the output of this function:
```julia
new_model, upper_variables, lower_variables, lower_objective = _build_single_model(model)
@test num_variables(new_model) == 2
@test length(upper_variables) == 1
@test length(lower_variables) == 1
@test objective_function(new_model) ≈ 3 * lower_variables[1] + upper_variables[1]
@test lower_objective ≈ -lower_variables[1]
# ... and so on
```

Reply:
Thanks, I was aware of this, I am playing around to learn what should I do.